javascript-style-guide.md

```diff
@@ -129,13 +165,13 @@ more productive by pointing out syntactic and logical errors before runtime.
 
 # Style
 
-## Leverage the functional programming paradigm.
+## Functionalism
 
-Prefer `forEach`, `map`, `filter` and `reduce` from `Array.prototype` over `for` loops.
+Prefer `forEach`, `map`, `filter` and `reduce` over `for` loops.
 
-Prefer `every` and `some` from `Array.prototype` over `continue` and `break`.
+Prefer `every` and `some` over `continue` and `break`.
 
-Prefer `Object.keys(object).forEach` over `for in` loops.
+Prefer iterating through own properties over `for in` loops.
```

Comment:
For a number of these conventions / recommendations (forEach, forOwn, isArray, etc), we've been relying on lodash. May as well formalize that here.